aries_cloudagent/protocols/present_proof/v2_0/formats/handler.py

```diff
@@ -0,0 +1,101 @@
+"""present-proof-v2 format handler - supports DIF and INDY."""
+from abc import ABC, abstractclassmethod, abstractmethod
+import logging
+
+from typing import Tuple
+
+from .....core.error import BaseError
+from .....core.profile import Profile
+from .....messaging.decorators.attach_decorator import AttachDecorator
+
+from ..message_types import ATTACHMENT_FORMAT
+from ..messages.pres import V20Pres
+from ..messages.pres_format import V20PresFormat
+from ..models.pres_exchange import V20PresExRecord
+
+LOGGER = logging.getLogger(__name__)
+
+PresFormatAttachment = Tuple[V20PresFormat, AttachDecorator]
+
+
+class V20PresFormatError(BaseError):
+    """Presentation exchange format error under present-proof protocol v2.0."""
+
+
+class V20PresFormatHandler(ABC):
+    """Base Presentation Exchange Handler."""
+
+    format: V20PresFormat.Format = None
+
+    def __init__(self, profile: Profile):
+        """Initialize PresExchange Handler."""
+        super().__init__()
+        self._profile = profile
+
+    @property
+    def profile(self) -> Profile:
+        """
+        Accessor for the current profile instance.
+
+        Returns:
+            The profile instance for this presentation exchange format
+
+        """
+        return self._profile
+
+    def get_format_identifier(self, message_type: str) -> str:
+        """Get attachment format identifier for format and message combination.
+
+        Args:
+            message_type (str): Message type for which to return the format identifier
+
+        Returns:
+            str: Issue credential attachment format identifier
+
+        """
+        return ATTACHMENT_FORMAT[message_type][self.format.api]
+
+    def get_format_data(self, message_type: str, data: dict) -> PresFormatAttachment:
+        """Get presentation format and attach objects for use in pres_ex messages."""
+        if self.format.api == "dif":
+            attach = AttachDecorator.data_json(data, ident=self.format.api)
```

Comment:
self.format is technically incorrect - it should be `V20PresFormatHandler.format`. Although python takes it promiscuously, this distinction can raise hell when trying to MagicMock around things in test code.

Reply:
The subclass extends `V20PresFormatHandler` and sets the format. If you use `V20PresFormatHandler` it would not distinct between the different formats. It can be either `IndyPresFormatHandler.format` or `DIFPresFormatHandler.format` (not the actual names, but something like that).

Reply:
I think my Python knowledge is lacking here. I've done some reading and get what you're saying here. 

This should also be fixed in the issue credential handler